app/controllers/concerns/authentication.rb

```diff
@@ -12,18 +12,22 @@ module Authentication
   ].freeze
 
   included do
-    before_action :authenticate_user
+    around_action :authenticate_user
   end
 
   def authenticate_user
+    if identity_header.blank?
+      return unauthenticated(error: 'Error parsing the X-RH-IDENTITY header')
+    end
     return unauthenticated unless identity_header.valid?
     return unauthorized unless rbac_allowed?
 
     return if performed?
 
     User.current = user
-  rescue JSON::ParserError, NoMethodError
-    unauthenticated error: 'Error parsing the X-RH-IDENTITY header'
+    yield
+  ensure
+    User.current = nil
```

Comment:
It might make sense to implement a `User.clear!`

Reply:
The name could be `User.clear_current!`. This concern seems to be the only exclusive place for changing its value, so the assignment should be fine.

Reply:
Yes, I agree that's more clear ;)